specs/src/specs/shares.md

```diff
@@ -0,0 +1,120 @@
+1. The number of shares in the matrix is a perfect square (see [tail padding share](#tail-padding-share))
+
+## Share Format
+
+The share format below is consistent for all blob shares. In other words, the share format below applies to shares with a namespace above [`MAX_RESERVED_NAMESPACE`](./consensus.md#reserved-namespaces) but below [`PARITY_SHARE_NAMESPACE`](./consensus.md#reserved-namespaces):
+
+- The first [`NAMESPACE_SIZE`](./consensus.md#constants) of a share's raw data is the namespace of that share.
+- The next [`SHARE_INFO_BYTES`](./consensus.md#constants) bytes are for share information with the following structure:
+  - The first 7 bits represent the share version in big endian form (initially, this will be `0000000` for version `0`);
+  - The last bit is a sequence start indicator. The indicator is `1` if the share is the first share of a sequence or `0` if the share is a continuation share of a sequence.
+- If this is the first share of a sequence the next [`SEQUENCE_BYTES`](./consensus.md#constants) contain a big endian `uint32` that represents the length of the sequence that follows in bytes.
+- The remaining [`SHARE_SIZE`](./consensus.md#constants)`-`[`NAMESPACE_SIZE`](./consensus.md#constants)`-`[`SHARE_INFO_BYTES`](./consensus.md#constants)`-`[`SEQUENCE_BYTES`](./consensus.md#constants) bytes (if first share) or [`SHARE_SIZE`](./consensus.md#constants)`-`[`NAMESPACE_SIZE`](./consensus.md#constants)`-`[`SHARE_INFO_BYTES`](./consensus.md#constants) bytes (if continuation share) are blob data. Note that blob data refers to the payload that user's submit in a [BlobTx](../../../x/blob/README.md).
+- If there is insufficient blob data to fill the share, the remaining bytes are filled with `0`.
+
+First share in a sequence:
+
+![fig: share start](./figures/share_start.svg)
+
+Continuation share in a sequence:
+
+![fig: share continuation](./figures/share_continuation.svg)
+
+Since blob data that exceeds [`SHARE_SIZE`](./consensus.md#constants)`-`[`NAMESPACE_SIZE`](./consensus.md#constants)`-`[`SHARE_INFO_BYTES`](./consensus.md#constants) `-` [`SEQUENCE_BYTES`](./consensus.md#constants) bytes will span more than one share, developers MAY choose to encode additional metadata in their raw blob data prior to inclusion in a Celestia block.
+
+## Transaction Shares
+
+In order for clients to parse shares in the middle of a sequence without downloading antecedent shares, Celestia encodes additional metadata in the shares associated with reserved namespaces. At the time of writing this only applies to the [`TRANSACTION_NAMESPACE`](./consensus.md#reserved-namespaces) and [`PAY_FOR_BLOB_NAMESPACE`](./consensus.md#reserved-namespaces). This share structure is often referred to as "compact shares" to differentiate from the share structure defined above for blob shares (a.k.a "sparse shares"). In other words, the format below applies to shares with below [`NAMESPACE_ID_MAX_RESERVED`](./consensus.md#reserved-namespaces):
```

Comment:
[Suggestion] I'd suggest making a connection between the field names in the figures and their textual description (similar to my comments for the "Share Format" section)

Reply:
Resolved by e035e9637783a9481d7eab5307addb06b6049fa0